### PR TITLE
DHT server initial implementation

### PR DESCRIPTION
**Pull request by kurnevsky**

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -1,3 +1,101 @@
...
+    let future = network_writer.select(network_reader).map(|_| ()).map_err(|(e, _)| e);
+    let future = future.select(server.run()).map(|_| ()).map_err(|(e, _)| e);
+    let future = future.select(lan_discovery_sender.run()).map(|_| ()).map_err(|(e, _)| e);
+    let future = future.map_err(|err| {
+        error!("Processing ended with error: {:?}", err);
+        ()
+    });
+
+    info!("Running server on localhost:33445");
+    tokio::run(future);
```
Comment (kpp) [inline src/main.rs]: Better use current_thread runtime.

File: src/main.rs [hunk 2]
```diff
@@ -1,3 +1,101 @@
...
+
+    let future = network_writer.select(network_reader).map(|_| ()).map_err(|(e, _)| e);
+    let future = future.select(server.run()).map(|_| ()).map_err(|(e, _)| e);
+    let future = future.select(lan_discovery_sender.run()).map(|_| ()).map_err(|(e, _)| e);
+    let future = future.map_err(|err| {
+        error!("Processing ended with error: {:?}", err);
+        ()
+    });
+
+    info!("Running server on localhost:33445");
```
Comment (kpp) [inline src/main.rs]: Log actual `local_addr`

File: Cargo.toml [hunk 3]
```diff
@@ -1,10 +1,19 @@
...
 repository = "https://github.com/tox-rs/tox-node/"
 license = "Unlicense"
 
 [dependencies]
+env_logger = "0.5"
+failure = "0.1"
+futures = "0.1"
+log = "0.4"
+tokio = "0.1.7"
+tox = { git = "https://github.com/tox-rs/tox", branch = "tox-node" }
```
Comment (kpp) [inline Cargo.toml]: changed branch to `tox-node`

File: src/main.rs [hunk 4]
```diff
@@ -1,3 +1,170 @@
...
+        .init();
+
+    if !crypto_init() {
+        panic!("Crypto initialization failed.");
+    }
+
+    let cli_config = cli_parse();
+
+    let local_addr: SocketAddr = "0.0.0.0:33445".parse().unwrap(); // 0.0.0.0 for ipv4
+    // let local_addr: SocketAddr = "[::]:33445".parse().unwrap(); // [::] for ipv6
```
Comment (NamsooCho) [inline src/main.rs]: How about switching IP version using command line argument

**Pull request merged**